smqtk_indexing/utils/metrics.py

```diff
@@ -97,19 +98,24 @@ def cosine_similarity(i: np.ndarray, j: np.ndarray) -> float:
     :param i: Vector i
     :param j: Vector j
 
-    :return: Float similarity.
+    :return: Float or array of cosine similarity.
     """
-    # numpy.linalg.norm is Frobenius norm (vector magnitude)
-    # return numpy.dot(i, j) / (numpy.linalg.norm(i) * numpy.linalg.norm(j))
+    if i.ndim == 1:
+        i = i.reshape(1, -1)
```

Comment:
Should we be asserting that `i.ndim == 1`? Otherwise the `[0]` indexing on line 109 will drop the remainder of a distance matrix computed of both `i` and `j` have `ndim > 1`.

Reply:
I went ahead and added an assert.